### PR TITLE
[MNT] Added `prettier` support

### LINKED ISSUE #11: Run `prettier` on pre-commit

### PR DESCRIPTION
**Pull request by samadpls**
- Closes #11 

Changes proposed in this pull request:

- Added support to Prettier on pre-commit

## Checklist

- [X] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
-  [X] PR links to GitHub issue with mention `Closes #11`
- [X] Tests pass
- [X] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Review by rmanaem — CHANGES_REQUESTED** [review]
Thanks for the PR @samadpls!
I think it's better to run `prettier` on `lint-staged` instead of directly running it on pre-commit. This ensures that only files/changes ready for commit will be formatted.
As for the title, may I suggest: [MNT] Configured prettier to run on lint-staged?
Let me know what you think.

File: .lintstagedrc.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "*.{ts,tsx}": ["npm run lint"]
+  "*.{ts,tsx}": [
+    "npm run lint",
+    "prettier --write",
+    "git add"
```
Comment (surchs) [inline .lintstagedrc.json]: Git add should remain a manual step

File: .lintstagedrc.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "*.{ts,tsx}": ["npm run lint"]
+  "*.{ts,tsx}": [
+    "npm run lint",
+    "prettier --write"
```
Comment (rmanaem) [inline .lintstagedrc.json]: It seems `prettier --write` doesn't run `prettier` as expected:

lint-staged

I suggest using `npm run format` instead.

**Review by rmanaem — CHANGES_REQUESTED** [review]

File: .lintstagedrc.json (hunk 2 above)
Comment (samadpls) [reply .lintstagedrc.json]: Thanks for letting me know. While reading the [documentation](https://btholt.github.io/complete-intro-to-react-v5/eslint-prettier#npmyarn-scripts), I also noticed that `npm run format` should be used for this.

**Review by rmanaem — APPROVED** [review]
Thanks for applying the requested changes @samadpls! I believe this is good to go 🧑‍🍳

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/query-tool/releases/tag/v0.1.0) :rocket:

**Pull request merged**